etl/steps/data/garden/animal_welfare/2023-08-01/global_hen_inventory.py

```diff
@@ -0,0 +1,132 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+from owid.catalog import Dataset, Table
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+# Columns to select from data, and how to rename them.
+COLUMNS = {
+    "country": "country",
+    "year": "year",
+    "barn": "share_of_hens_in_barns",
+    "brown__pct": "share_of_brown_hens",
+    "cage": "share_of_hens_in_cages",
+    "cage_free": "number_of_hens_cage_free",
+    "cages": "number_of_hens_in_cages",
+    "commercial_egg_farms": "number_of_commercial_egg_farms",
+    "free_range": "share_of_hens_free_range_not_organic",
+    "organic": "share_of_hens_free_range_organic",
+    "total_layers": "number_of_laying_hens",
+    "unknown": "share_of_hens_in_unknown_housing",
+    # The following columns will be used to extract sources and urls and add them to the source metadata description.
+    # Afterwards, they will be removed.
+    "available_at": "available_at",
+    "source": "source",
+    # "click_placements" : "click_placements",
+    # "number_of_records": "number_of_records",
+}
+
+
+def add_individual_sources_to_metadata(tb: Table) -> Table:
+    tb = tb.copy()
+    # Check that each country has only one source.
+    assert (tb.groupby("country").agg({"source": "nunique"})["source"] == 1).all()
```

Comment:
Could you add an error message to make this assertion more useful? E.g. "There are countries with more than one source! Only one source per country is expected!".